cache.go

```diff
@@ -5,17 +5,26 @@ import (
 	"time"
 )
 
+// CachedValue is a type for cache value
+type CachedValue interface{}
+
+// CachedItems is a map for CachedValue
+type CachedItems map[string]CachedValue
```

Comment:
We don't need the `CachedItems` type since changing `map[string]...` would break any of the functions using a `key string`. It's not something that could be changed very easily.